app/services/search/location_builder.rb

```diff
@@ -25,6 +25,7 @@ def initialize(location, radius, location_category, buffer_radius)
     elsif location_category_search?
       initialize_location_polygon
     elsif @location.present?
+      map_location_with_incorrect_coordinates if MAPPED_POINT_LOCATIONS.include?(@location)
```

Comment:
It would be safer to do the reassignment of `@location` earlier in the process. It's already been referenced 3 times before this line. That happens to be OK at the moment but it might not always be ok. Let's map it as soon as we assign it for the first time (unless there is a reason we shouldn't?)